### PR TITLE
feat: automatically set region based on neighborhood

### LINKED ISSUE #1294: Detroit - Add default to region
As a partner in Detroit, 
I want the region to be auto-populated 
So that it's easier for me to add listings 

**Acceptance Criteria**
* When I select a neighborhood from the dropdown, then the region should automatically default based off of the following mapping below 

https://docs.google.com/spreadsheets/d/15KIX_wSuKmrtjYYrgRgb-LN1cdxIIJ9RSffPJ7SQ7j4/edit#gid=1612299484

### PR DESCRIPTION
**Pull request by emilyjablonski**
## Issue

- Closes #1294 

## Description

Automatically selects the mapped region when a neighborhood value is selected in the listing form.

Also moved two files that had to be touched `filters` and `region` from ui-components into shared-helpers due to backend dependencies.

## How Can This Be Tested/Reviewed?

In the listing form, change the neighborhood value and the region value should populate with its mapped value. Ensure you can change it manually from its mapped value, save it, and that it re-populates with the custom value.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: shared-helpers/src/regions.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
+export enum Region {
```
Comment (emilyjablonski) [inline shared-helpers/src/regions.ts:1]: Just moved this file out of ui-components, the diff didn't follow

File: shared-helpers/src/filters.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@ import {
   ListingFilterKeys,
```
Comment (emilyjablonski) [inline shared-helpers/src/filters.ts:3]: Moved out of ui-components because of backend references

File: shared-helpers/src/regions.ts (hunk 1 above)
Comment (seanmalbert) [reply shared-helpers/src/regions.ts:1]: Thank you for these comments, they're helpful!

**Review by seanmalbert — APPROVED** [review]
lgtm!

**Pull request merged**